sanitize.go

```diff
@@ -0,0 +1,57 @@
+package scroll
+
+// The AllowSet interface is implemented to detect if input is safe or not.
+type AllowSet interface {
+	IsSafe(string) bool
+}
+
+// AllowSetChars allows the definition of a set of safe allowed characters.
+type AllowSetChars struct {
+	maxLen int
+	chars  [256]bool
+}
+
+func NewAllowSetChars(s string, maxlen int) AllowSetChars {
```

Comment:
I'm a little worried about someone calling `NewAllowSetChars("über", 10)` and then being surprised when it validates the string `"�"`. Changing the name to `NewAllowSetBytes` will help with that. I propose also adding a comment, like: `AllowSetBytes does not know about unicode code points. If you pass the string "ü" (which encodes as 0xc3 0xbc) it will allow the bytes 0xc3 and 0xbc separately, which means it will accept strings like "\xc3\xc3\xc3\xc3\xc3\xc3". This may not be what you intend.`